src/android/Geolocation.java

```diff
@@ -185,14 +258,45 @@ private void addWatch(LocationContext locationContext) {
             request.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
         }
 
-        requestLocationUpdatesIfSettingsSatisfied(locationContext, request);
+        requestLocationUpdatesIfSettingsSatisfiedGms(locationContext, request);
+    }
+
+    /**
+     * Add Watch Locations in devices which doesn't contains google services (Huawei Devices)
+     * @param locationContext the locationContext
+     */
+    private void addWatchHuawei(LocationContext locationContext) {
+        JSONArray args = locationContext.getExecuteArgs();
+        boolean enableHighAccuracy = args.optBoolean(1, false);
+        long maximumAge = args.optLong(2, 5000);
+
+        com.huawei.hms.location.LocationRequest request = com.huawei.hms.location.LocationRequest.create();
+
+        request.setInterval(maximumAge);
+
+        if(enableHighAccuracy) {
+            request.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
+        }
+
+        requestLocationUpdatesIfSettingsSatisfiedHms(locationContext, request);
     }
 
     @SuppressLint("MissingPermission")
-    private void requestLocationUpdates(LocationContext locationContext, LocationRequest request) {
-        fusedLocationClient.requestLocationUpdates(request, locationContext.getLocationCallback(), null);
+    private void requestLocationUpdatesGms(LocationContext locationContext, LocationRequest request) {
+        fusedLocationClientGms.requestLocationUpdates(request, locationContext.getLocationCallback(), null);
     }
 
+    @SuppressLint("MissingPermission")
+    private void requestLocationUpdatesHms(LocationContext locationContext, com.huawei.hms.location.LocationRequest request) {
+        if (fusedLocationProviderClientHms != null) {
+            fusedLocationProviderClientHms.requestLocationUpdates(request, locationContext.getLocationCallbackHuawei(),  Looper.getMainLooper());
+        }
+    }
+
+    /**
+     * Clear Watch Locations in devices which contains google services
+     * @param callbackContext the callbackContext
+     */
     private void clearWatch(JSONArray args, CallbackContext callbackContext) {
```

Comment:
In order to follow the same approach as other methods, shouldn't this be changed to `clearWatchGms`?

However, most of the code inside the `clearWatch/clearWatchHms` methods is identical besides one or two property/methods. Can't we reuse code as much as possible in order to avoid too much replication?